db/config.cc

```diff
@@ -1165,4 +1170,74 @@ future<gms::inet_address> resolve(const config_file::named_value<sstring>& addre
     co_return coroutine::exception(std::move(ex));
 }
 
+static future<std::vector<seastar::metrics::relabel_config>> get_relable_from_file(const std::string& name) {
+    return seastar::open_file_dma(name, open_flags::ro).then([](file f) {
+        return f.size().then([f](size_t s) {
+            return do_with(seastar::make_file_input_stream(f), [s](seastar::input_stream<char>& in) {
+                return in.read_exactly(s).then([](temporary_buffer<char> buf) {
```

Comment:
You could use coroutines here, it would greatly improve readability.

Reply:
sure